src/arizona_template_compiler.erl

```diff
@@ -0,0 +1,653 @@
+-module(arizona_template_compiler).
+-moduledoc false.
+
+%% --------------------------------------------------------------------
+%% API function exports
+%% --------------------------------------------------------------------
+
+-export([compile/3]).
+
+%
+
+-ignore_xref([compile/3]).
+
+%% --------------------------------------------------------------------
+%% Types (and their exports)
+%% --------------------------------------------------------------------
+
+-record(state, {
+    module :: module(),
+    function :: atom(),
+    macros :: assigns(),
```

Comment:
Is this the same as `arizona_live_view:macros()`? If so one of them should disappear and/or be renamed 😄

Reply:
Yes, they are the same. What do you suggest?

Reply:
Maybe create an issue for this situation? Then we can resolve it async.

Reply:
We can solve it now, since `assigns` is exposed, right? Just remove the other type, by replacing it with this one. If you run into Dialyzer/opacity-related issues, just make it a regular type, for now.

Reply:
[a063ed3](https://github.com/arizona-framework/arizona/pull/164/commits/a063ed3b3224ab48438459655faf7529a52bcea0)